packages/react/src/components/Toast/toast.test.tsx

```diff
@@ -0,0 +1,288 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import Toast, { type ToastProps } from './';
+import axe from '../../axe';
+import userEvent from '@testing-library/user-event';
+import AriaIsolate from '../../utils/aria-isolate';
+
+const testString = 'a simple paragraph';
+
+const toastTypes: Record<ToastProps['type'], string> = {
+  confirmation: 'success',
+  'action-needed': 'error',
+  error: 'error',
+  info: 'info',
+  caution: 'warning'
+};
+
+const toastIcons: Record<ToastProps['type'], string> = {
+  confirmation: 'check-circle',
+  'action-needed': 'no',
+  error: 'caution',
+  info: 'info-circle-alt',
+  caution: 'caution'
+};
+
+afterEach(() => {
+  jest.restoreAllMocks();
+});
+
+Object.entries(toastTypes).forEach(([key, value]) => {
+  test(`should correctly render toast with type="${value}" when dismissed, but mounted`, async () => {
+    render(
+      <Toast
+        show={false}
+        type={key as keyof typeof toastTypes}
+        data-testid="toast"
+      >
+        {testString}
+      </Toast>
+    );
+    const toast = screen.getByTestId('toast');
+    expect(toast).toHaveClass(`Toast Toast--${value} is--hidden`);
+    expect(toast).toHaveTextContent(testString);
+  });
+
+  test(`should include the correct classes with a type="${value}" toast and show is true`, async () => {
+    render(
+      <Toast type={key as keyof typeof toastTypes} show data-testid="toast">
+        {testString}
+      </Toast>
+    );
+    const toast = screen.getByTestId('toast');
+    expect(toast).toBeInTheDocument();
+    expect(toast).toHaveClass(`Toast Toast--${value} FadeIn--flex`);
+  });
+
+  test('should transition from hidden to shown', async () => {
+    let show = false;
+
+    render(
+      <Toast type="info" show={show} data-testid="toast">
+        {testString}
+      </Toast>
+    );
+    const toast = screen.getByTestId('toast');
+    expect(toast).toHaveClass('Toast Toast--info is--hidden');
+    show = true;
+    setTimeout(() => {
+      expect(toast).toHaveClass('Toast Toast--info FadeIn--flex');
+    }, 300);
```

Comment:
So the RTL way we've been doing this and I've seen recommended is to use [`rerender`](https://testing-library.com/docs/example-update-props/), example here: https://github.com/dequelabs/cauldron/blob/b25e7c3390360af6e5261af41b120b56e87012ef/packages/react/src/components/Select/index.test.tsx#L130